plugins/UserId/Reports/GetUsers.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\UserId\Reports;
+
+use Piwik\Piwik;
+use Piwik\Plugin\Report;
+use Piwik\Plugin\ViewDataTable;
+use Piwik\Plugins\UserId\Visualizations\UserIds;
+use Piwik\View;
+
+/**
+ * A report showing all unique user IDs and some aggregated information about them. It also allows
+ * to open a popover with visitor details
+ */
+class GetUsers extends Base
+{
+    /**
+     * @return array
+     */
+    public static function getColumnsToDisplay()
+    {
+        return array(
+            'label', 'nb_visits', 'nb_actions', 'nb_visits_converted'
+        );
+    }
+
+    protected function init()
+    {
+        parent::init();
+
+        $this->name          = Piwik::translate('UsersManager_MenuUsers');
+        $this->menuTitle     = $this->name;
+        $this->documentation = '';
+
+        // This defines in which order your report appears in the mobile app, in the menu and in the list of widgets
+        $this->order = 1;
+    }
+
+    /**
+     * @param ViewDataTable $view
+     */
+    public function configureView(ViewDataTable $view)
+    {
+        $view->config->addTranslation('label', Piwik::translate('General_UserId'));
+        $view->config->addTranslation('nb_visits_converted', Piwik::translate('General_VisitConvertedGoal'));
+
+        /*
+         * Hide most of the table footer actions, leaving only export icons and pagination
+         */
+        $view->config->columns_to_display = $this->getColumnsToDisplay();
+        $view->config->show_all_views_icons = false;
+        $view->config->show_active_view_icon = false;
+        $view->config->show_exclude_low_population = false;
+        $view->config->show_export_as_rss_feed = false;
+        $view->config->show_related_reports = false;
+        $view->config->show_insights = false;
+        $view->config->show_pivot_by_subtable = false;
+        $view->config->show_flatten_table = false;
+        $view->config->show_table = false;
+        $view->config->show_table_all_columns = false;
+        $view->config->disable_row_evolution = true;
+    }
+
+    public function getDefaultTypeViewDataTable()
+    {
+        return UserIds::ID;
```

Comment:
>  I'm using a custom visualization here because I'd like to pass a custom visitor_url metadata to HTML attribute

Good point. I thought about a solution for this for a while and maybe we could instead add an attribute `data-row-metadata` here https://github.com/piwik/piwik/blob/2.16.0/plugins/CoreVisualizations/templates/_dataTableViz_htmlTable.twig#L35 which contains the JSON encoded metadata of a row. For example

```php
data-row-metadata="{{ row.getMetadata|json_encode|e('html_attr') }}"
```

similar to how it is already done for the table metadata see https://github.com/piwik/piwik/blob/2.16.0/plugins/CoreHome/templates/_dataTable.twig#L10 . 

This way we would not need the custom visualization maybe. It would be simply easier to maintain and reuse one visualization instead of two and whenever we have a similar case we won't need to create a third table visualization. It otherwise just requires more work afterwards, more potential security risks etc. 

> BTW I noticed that columns names (e.g. label) are used in conditions both in logic and view. This brings an unneeded dependency on column names, that usually is not obvious

I would say the `label` column is ok to treat it kind of special and to have it in both logic and view since all dimension reports are pretty much required to have a column `label`. Would be good to not have special logic for other metadata and columns like `url` and `html_label_prefix` and instead use the `data-row-metadata` in the future maybe. We want to refactor and improve this when moving the visualizations from Twig to Angular and then render all visualizations client side. That's why we won't make too many changes for this currently.